### PR TITLE
[BEAM-7060] Introduce Python3-only test modules

### PR DESCRIPTION
**Pull request by udim**
This is required for testing type annotations support (PEP 484).

Note that `*py3.py` files are only intended for **tests** where the **syntax** is incompatible with Py2. Not intended to replace checks for where certain versions of Python behave differently.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by udim** [comment]
R: @tvalentyn @chamikaramj

File: sdks/python/scripts/run_pylint.sh [hunk 1]
```diff
@@ -60,23 +60,34 @@ EXCLUDED_GENERATED_FILES=(
 apache_beam/portability/api/*pb2*.py
 )
 
+PYTHON_MAJOR=$(python -c 'import sys; print(sys.version_info[0])')
```
Comment (chamikaramj) [inline sdks/python/scripts/run_pylint.sh]: Please add a JIRA to do lint checks for Python 3.

File: sdks/python/scripts/run_pylint.sh [hunk 2]
```diff
@@ -60,23 +60,34 @@ EXCLUDED_GENERATED_FILES=(
...
+if [[ "${PYTHON_MAJOR}" == 2 ]]; then
+  EXCLUDED_PY3_FILES=$(find ${MODULE} | grep 'py3\.py$')
+  echo -e "Excluding Py3 files:\n${EXCLUDED_PY3_FILES}"
+else
+  EXCLUDED_PY3_FILES=""
+fi
+
 FILES_TO_IGNORE=""
-for file in "${EXCLUDED_GENERATED_FILES[@]}"; do
+for file in "${EXCLUDED_GENERATED_FILES[@]}" ${EXCLUDED_PY3_FILES}; do
```
Comment (tvalentyn) [inline sdks/python/scripts/run_pylint.sh]: Does this loop work as intended when $EXCLUDED_PY3_FILES has more than one file?

Alternative solution: https://github.com/apache/beam/pull/8505/files#diff-d91ac373b8d2235683a85576912b5db5R66
which requires: https://github.com/apache/beam/pull/8505/files#diff-d91ac373b8d2235683a85576912b5db5R87

**Comment by tvalentyn** [comment]
Run Python 2 PostCommit

File: sdks/python/scripts/run_pylint.sh (hunk 1 above)
Comment (udim) [reply sdks/python/scripts/run_pylint.sh]: done: https://issues.apache.org/jira/browse/BEAM-7884

File: sdks/python/scripts/run_pylint.sh (hunk 2 above)
Comment (udim) [reply sdks/python/scripts/run_pylint.sh]: Yes, see: https://scans.gradle.com/s/544uyirsdn6n4/console-log#L206-L209

**Comment by udim** [comment]
@tvalentyn I believe pre-commits cover all the changes in this PR.

**Comment by tvalentyn** [comment]
A similar change was reverted because it broke postcommits:  https://github.com/apache/beam/pull/8505#issuecomment-498441270

**Comment by udim** [comment]
run python precommit

**Comment by udim** [comment]
run python 2 postcommit

**Comment by udim** [comment]
Run Python Dataflow ValidatesContainer

**Comment by udim** [comment]
Error in postcommit is a transient pip issue

**Comment by chamikaramj** [comment]
LGTM. Thanks.

**Comment by amaliujia** [comment]
Hello!

I am thinking if this PR causes `./gradlew spotlessApply` fail in master branch.

Run `./gradlew spotlessApply` gives the following error message on master branch:

```
* Where:
Build file '/Users/ruwang/Documents/beam/sdks/python/apache_beam/testing/load_tests/build.gradle' line: 44

* What went wrong:
A problem occurred evaluating project ':sdks:python:apache_beam:testing:load_tests'.
> Could not get unknown property 'files' for task ':sdks:python:apache_beam:testing:load_tests:run' of type org.gradle.api.tasks.Exec.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
```

**Comment by tvalentyn** [comment]
Thanks, @amaliujia, I opened https://issues.apache.org/jira/browse/BEAM-7892 to track this.

**Pull request merged**